### PR TITLE
Support to update quota in pod move actions

### PR DESCRIPTION
**Pull request by irfanurrehman**
Fixes https://vmturbo.atlassian.net/browse/OM-65374

### CODE REVIEW

File: test/integration/pod_move_with_quota.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	kubeclientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	restclient "k8s.io/client-go/rest"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/turbonomic/kubeturbo/pkg/action"
+	"github.com/turbonomic/kubeturbo/pkg/cluster"
+)
+
+var _ = Describe("Action Execution when quota exists ", func() {
```
Comment (ading1977) [inline test/integration/pod_move_with_quota.go:23]: Does the integration test validate that the temporary quota annotation is removed and the quota value is reverted  after the pod move is finished?

File: pkg/action/executor/quota_util.go [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+			return err
+		}
+		if !match {
+			continue
+		}
+
+		hardResources := quota.ResourceNames(resourceQuota.Status.Hard)
+		restrictedResources := q.evaluator.MatchingResources(hardResources)
+
+		if !hasUsageStats(&resourceQuota, restrictedResources) {
```
[the remark this answers is not in the document]
Comment (irfanurrehman) [reply pkg/action/executor/quota_util.go:124]: Thanks @ading1977 for catching this. What I did intend to do was to skip the quota check if there is no usage stat for any of the quotas under consideration. Updated accordingly!

File: pkg/action/executor/quota_util.go [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+			glog.Warningf("No usage stats found on resource quota: %s, while validating quota for pod: %s.", resourceQuota.Name, podQualifiedName)
+			continue
+		}
+
+		requestedUsage := quota.Mask(deltaUsage, hardResources)
+		newTotalUsage := quota.Add(resourceQuota.Status.Used, requestedUsage)
+		// We need to mask again as add returns all resources in the first list
+		maskedNewTotalUsage := quota.Mask(newTotalUsage, quota.ResourceNames(requestedUsage))
+
+		if allowed, exceeded := quota.LessThanOrEqual(maskedNewTotalUsage, resourceQuota.Status.Hard); !allowed {
```
[the remark this answers is not in the document]
Comment (irfanurrehman) [reply pkg/action/executor/quota_util.go]: Updated!

File: pkg/action/executor/quota_util.go [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+		}
+		_, err = q.client.CoreV1().ResourceQuotas(q.namespace).Update(context.TODO(), revertedQuota, metav1.UpdateOptions{})
+		if err != nil {
+			glog.Warningf("Error reverting the updated quota: %s/%s: %v", quota.Namespace, quota.Name, err)
+			continue
+		}
+	}
+}
+
+func getScopeSelectorsFromQuota(quota corev1.ResourceQuota) []corev1.ScopedResourceSelectorRequirement {
```
[the remark this answers is not in the document]
Comment (irfanurrehman) [reply pkg/action/executor/quota_util.go]: Removed!

File: test/integration/pod_move_with_quota.go (hunk 1 above)
Comment (irfanurrehman) [reply test/integration/pod_move_with_quota.go:26]: It did not, but have added the same now. Have put the changes in a saperate commit for clear reference.

**Comment by irfanurrehman** [comment]
Thanks @ading1977 for the review. Have updated, PTAL!

**Review by ading1977 — APPROVED** [review]
LGTM

**Comment by irfanurrehman** [comment]
Merging this to clear the commit backlog from the forthcoming PRs, which are work on top.

**Pull request merged**